File: DataStructures/Queue/Java/QueueTest.java
public class QueueTest {

	public static void main(String[] args) {
		Queue queue = new Queue();
		System.out.println(queue);
		try {
			queue.dequeue();
		} catch (Exception e) {
			System.out.println(e.getMessage());
		}
		System.out.println(queue.isEmpty());
		System.out.println(queue.getSize());
		queue.enqueue(80);
		queue.enqueue(890);
		queue.enqueue(-9);
		System.out.println(queue.getSize());
		try {
			System.out.println(queue.peek());
		} catch (Exception e) {
			System.out.println(e.getMessage());
		}
		System.out.println(queue);
		System.out.println(queue);
		while (!queue.isEmpty()) {
			try {
				System.out.println(queue.getSize() + "\t" + queue.dequeue() + "\t" + queue.getSize());
			} catch (Exception e) {
				System.out.println(e.getMessage());
			}
		}
		
		System.out.println(queue.isEmpty());

		for (int i = 0; i < 10; i++) {
			queue.enqueue(i);
		}

		while (!queue.isEmpty()) {
			try {
				System.out.println(queue.getSize() + "\t" + queue.dequeue() + "\t" + queue.getSize() + "\t" + queue.isEmpty());
			} catch (Exception e) {
				System.out.println(e.getMessage());
			}
		}

		System.out.println(queue);
	}
}

File: DataStructures/Queue/JavaScript/QueueTest.js
require('./Queue');

var q = new Queue();
q.print();
console.log(q.isEmpty());

q.enqueue(7);
q.enqueue("second");
q.enqueue([6, 8]);

q.print();
console.log(q.isEmpty());
console.log(q.dequeue());
console.log(q.dequeue());
console.log(q.dequeue());
q.print();
console.log(q.dequeue());
console.log(q.isEmpty());
q.enqueue(88);
q.enqueue(99);
q.enqueue([6, 8]);
q.print();
console.log(q.isEmpty());
console.log(q.clear())
console.log(q.isEmpty());
q.print();
q.enqueue(88);
q.enqueue(99);
q.enqueue([6, 8]);
console.log(q.dequeue());
console.log(q.dequeue());
console.log(q.dequeue());
q.print();
